### DISCUSSION TITLE
Need Help Storing Complete AIS Data in a Database

### DISCUSSION THREAD

**Discussion by Parabellum123** [jvde-github/AIS-catcher#299] [Q&A]
Hello everyone,

I am currently working on a project involving the collection of AIS (Automatic Identification System) data using a receiver and storing it in a PostgreSQL database. The data I receive includes various information about ships, such as MMSI, location, speed, and more.

The problem I’m facing is that currently, only the last information received from each ship (based on MMSI) is stored in the database. I want every signal received, even if it’s from the same ship within a short time span, to be stored as a new entry in the database. The goal is to ensure that I have a complete record of all the data received, including multiple signals from the same ship within close time intervals.

Could anyone advise on what changes I need to make, either in my Python code or in the PostgreSQL database configuration, to allow this?

Thank you for your help!

my postgresql
![image](https://github.com/user-attachments/assets/5f7da532-4ab5-4c93-aadb-4778414ccc31)

i want like this
![image](https://github.com/user-attachments/assets/5d501938-a860-4941-b429-3baa746d03d3)

**Comment by jvde-github** [comment]
Hello!

If you are using the built-in PSQL functionality (i.e. with the -D) you can use 'VP on` which essentially populates the table ais_vessel_pos with lat/lon with some additional fields including timestamp, mmsi and speed.

If you want to use a custom script in Python, then you need to read in NMEA lines (the messages) (e.g. by implementing a UDP listener), decoding them in python (e.g. with pyais to get the lat and lon) and then writing this to the table. All together, not rocket science but requires a bit of work.

**Reply by Parabellum123** [reply]
1
![image](https://github.com/user-attachments/assets/c3914d30-c736-42f4-b833-74f1d4d41827)
2
![image](https://github.com/user-attachments/assets/3d94d5b2-9f56-4086-840a-4dedf830c97e)

in the first photo i can run ais catcher and send it to the data base
but when i turn on the rest i can't run ais catcher, is there something wrong in my python script?
i follow as in https://github.com/jvde-github/AIS-catcher?tab=readme-ov-file#writing-ais-messages-to-a-postgres-database

i have also tried with "-VP ON" "-VS ON" ....

**Reply by Parabellum123** [reply]
![image](https://github.com/user-attachments/assets/f39f05ac-eca0-4a33-ad7b-3d1778897425)

this is the same in cmd, it doesn't work when i want on the other table

**Reply by Parabellum123** [reply]
![image](https://github.com/user-attachments/assets/fa1b7d29-f0be-42c0-b373-1d2f316e841b)

Buut if i do it the normal database it work

**Reply by jvde-github** [reply]
Sorry, the VP on is part of the -D command (it is an option for the database functionality activated with -D). So these options need to be after the -D but before the next "-"

Hence, 
```
-D postgressql://.... VP on
```
Also, -V -> -v. Does that help?

**Reply by jvde-github** [reply]
So, I suspect that at the end of the "-D" line in your python, you add: ,"VP on" as an extra argument

**Reply by jvde-github** [reply]
well, probably add it as: ,"VP","ON"

**Reply by Parabellum123** [reply]
![image](https://github.com/user-attachments/assets/2da8b0de-215a-44e8-bf67-c18cbae1aa7f)

OMG It work, thank you very much

**Reply by jvde-github** [reply]
Have fun with it and thanks for sharing feed.
